### PR TITLE
V4 cleanup psalm

### PR DESCRIPTION
**Pull request by DavidGoodwin**
I added vimeo/psalm to a local composer.json and ran it over the codebase. 

These were the seemingly easy things to fix.

### CODE REVIEW

File: src/Command/PeekJobCommand.php [hunk 1]
```diff
@@ -42,14 +39,7 @@ public function getCommandLine()
...
     {
-        if ($responseLine == ResponseInterface::RESPONSE_NOT_FOUND) {
-            $message = sprintf(
-                '%s: Job %u does not exist.',
-                $responseLine,
-                $this->jobId
-            );
-            throw new Exception\ServerException($message);
-        } elseif (preg_match('#^FOUND (\d+) \d+$#', $responseLine, $matches)) {
+        if (preg_match('#^FOUND (\d+) \d+$#', $responseLine, $matches)) {
```
Comment (SamMousa) [inline src/Command/PeekJobCommand.php]: What's the reason for swapping these checks?
Checking for an exact match is faster so it makes sense to do that test first...

File: src/Socket/NativeSocket.php [hunk 2]
```diff
@@ -57,7 +57,7 @@ public function write($data)
     {
         $history = new WriteHistory(self::WRITE_RETRIES);
 
-        for ($written = 0, $fwrite = 0; $written < strlen($data); $written += $fwrite) {
+        for ($written = 0; $written < strlen($data); $written += $fwrite) {
```
Comment (SamMousa) [inline src/Socket/NativeSocket.php]: This looks like it changes behavior..

File: src/Command/PeekJobCommand.php (hunk 1 above)
Comment (DavidGoodwin) [reply src/Command/PeekJobCommand.php]: I changed it when I removed the elseif; order is reverted now.

**Comment by SamMousa** [comment]
@DavidGoodwin tests seem to fail; you up for fixing it? (Possibly need to merge my PR into yours again)

**Pull request merged**